includes/TripalFields/schema__itemlocation/schema__itemlocation_widget.inc

```diff
@@ -75,6 +75,10 @@ class schema__itemlocation_widget extends ChadoFieldWidget {
         $query->distinct();
         $query->condition('FM.uri', $uri);
         $local = $query->execute()->fetchField();
+        // we can also use the public:// prefix with local files not managed by tripal
+        if (!$local) {
```

Comment:
Rather then check if $local is empty here, how about adding another if statement on line 71:

```
if (preg_match('/^public:\/\//', $uri) {
  remote = $uri
}
```

Reply:
I didn't do it that way because tripal managed files can also have a public:// prefix, e.g. ```public://tripal/users/1/T119314.tree```
As best as I could figure, I need to see if it is managed first.

Reply:
There are new changes, see what you think